kola/harness.go

```diff
@@ -418,16 +418,30 @@ func RunTest(t *register.Test, pltfrm string) error {
 		}
 	}
 
-	// run test
-	err = t.Run(tcluster)
-
 	// give some time for the remote journal to be flushed so it can be read
 	// before we run the deferred machine destruction
-	if err != nil {
-		time.Sleep(10 * time.Second)
-	}
+	defer func() {
+		if err != nil {
+			time.Sleep(2 * time.Second)
+		}
+	}()
+
+	defer func() {
+		if r := recover(); r != nil {
+			switch r := r.(type) {
```

Comment:
`The TypeSwitchGuard may include a short variable declaration. When that form is used, the variable is declared at the beginning of the implicit block in each clause. In clauses with a case listing exactly one type, the variable has that type; otherwise, the variable has the type of the expression in the TypeSwitchGuard.` - spec

`If the switch declares a variable in the expression, the variable will have the corresponding type in each clause. It's also idiomatic to reuse the name in such cases, in effect declaring a new variable with the same name but a different type in each case.` - effective go

without the short variable declaration, pretty sure r is still interface{}.